tests/src/JIT/HardwareIntrinsics/Arm/Shared/GenerateTests.csx

```diff
@@ -70,6 +70,24 @@ private static readonly (string templateFileName, Dictionary<string, string> tem
     ("SimpleBinOpTest.template",  new Dictionary<string, string> { ["TestName"] = "Add_Vector128_UInt64",             ["Isa"] = "AdvSimd", ["LoadIsa"] = "AdvSimd", ["Method"] = "Add",                                      ["RetVectorType"] = "Vector128", ["RetBaseType"] = "UInt64",  ["Op1VectorType"] = "Vector128", ["Op1BaseType"] = "UInt64", ["Op2VectorType"] = "Vector128", ["Op2BaseType"] = "UInt64", ["LargestVectorSize"] = "16", ["NextValueOp1"] = "TestLibrary.Generator.GetUInt64()",        ["NextValueOp2"] = "TestLibrary.Generator.GetUInt64()", ["ValidateFirstResult"] = "(ulong)(left[0] + right[0]) != result[0]",                                                                                           ["ValidateRemainingResults"] = "(ulong)(left[i] + right[i]) != result[i]"}),
     ("SimpleBinOpTest.template",  new Dictionary<string, string> { ["TestName"] = "AddScalar_Vector64_Single",        ["Isa"] = "AdvSimd", ["LoadIsa"] = "AdvSimd", ["Method"] = "AddScalar",                                ["RetVectorType"] = "Vector64",  ["RetBaseType"] = "Single",  ["Op1VectorType"] = "Vector64",  ["Op1BaseType"] = "Single", ["Op2VectorType"] = "Vector64",  ["Op2BaseType"] = "Single", ["LargestVectorSize"] = "8",  ["NextValueOp1"] = "TestLibrary.Generator.GetSingle()",        ["NextValueOp2"] = "TestLibrary.Generator.GetSingle()", ["ValidateFirstResult"] = "BitConverter.SingleToInt32Bits(left[0] + right[0]) != BitConverter.SingleToInt32Bits(result[0])",                                    ["ValidateRemainingResults"] = "BitConverter.SingleToInt32Bits(result[i]) != 0"}),
+    ("BitUnOpTest.template",      new Dictionary<string, string> { ["TestName"] = "LeadingSignCount_Vector64_Int16",  ["Isa"] = "AdvSimd", ["LoadIsa"] = "AdvSimd", ["Method"] = "LeadingSignCount",                         ["RetVectorType"] = "Vector64",  ["RetBaseType"] = "Int16",   ["Op1VectorType"] = "Vector64",  ["Op1BaseType"] = "Int16",                                                               ["LargestVectorSize"] = "8",  ["NextValueOp1"] = "(short)-TestLibrary.Generator.GetInt16()",                                                         ["ValidateFirstResult"] = "CountLeadingSignBits(firstOp[0]) != result[0]",                                                                                      ["ValidateRemainingResults"] = "CountLeadingSignBits(firstOp[i]) != result[i]"}),
```

Comment:
This could just be done via `_UnaryOpTestTemplate.template` and having a different validation logic per scenario (`CountLeadingSignBits`, `CountLeadingZeroBits`, etc).

Reply:
@tannergooding Can you please take another look ?

As we discussed privately, I have moved `CountLeadingSignBits`, `CountLeadingZeroBits`, `BitCount` functions' implementations into T4 template file Helpers.tt and included the corresponding Helpers.cs in the generated test project files. 

This way we can keep adding functions needed for validation to a separate file without increasing the number of templates and without inlining them into GenerateTests.csx.